runtime/gc_glue_java/MarkingDelegate.cpp

```diff
@@ -261,18 +261,19 @@ void
 MM_MarkingDelegate::scanContinuationNativeSlots(MM_EnvironmentBase *env, omrobjectptr_t objectPtr)
 {
 	J9VMThread *currentThread = (J9VMThread *)env->getLanguageVMThread();
-	if (VM_VMHelpers::needScanStacksForContinuation(currentThread, objectPtr)) {
+	if (VM_VMHelpers::needScanStacksForContinuation(currentThread, objectPtr, true)) {
 		StackIteratorData4MarkingDelegate localData;
 		localData.markingDelegate = this;
 		localData.env = env;
 		localData.fromObject = objectPtr;
 
-		bool bStackFrameClassWalkNeeded = false;
+		bool stackFrameClassWalkNeeded = false;
 #if defined(J9VM_GC_DYNAMIC_CLASS_UNLOADING)
-		bStackFrameClassWalkNeeded = isDynamicClassUnloadingEnabled();
+		stackFrameClassWalkNeeded = isDynamicClassUnloadingEnabled();
 #endif /* J9VM_GC_DYNAMIC_CLASS_UNLOADING */
 
-		GC_VMThreadStackSlotIterator::scanSlots(currentThread, objectPtr, (void *)&localData, stackSlotIteratorForMarkingDelegate, bStackFrameClassWalkNeeded, false);
+		bool syncWithConcurrentGCScan = J9_ARE_ANY_BITS_SET(currentThread->privateFlags, J9_PRIVATE_FLAGS_CONCURRENT_MARK_ACTIVE);
```

Comment:
Hm, sorry, I was wrong about the name of the flag... More appropriate is syncWithContinuationMounting. This Thread is trying to do GC work, and it races with the other one doing mounting, and the question is if we do or do not need to sync with them.
All the flag names on this call sequence should be renamed accordingly.

On the other side, method synchronizeWithConcurrentGCScan is appropriately named because it is called in the context of carrier thread doing mount and needing to sync with racing GC thread. 

An also add a comment above the flag something like:
In STW GC there are not no racing carrier threads doing mount and no need for the synchronization.

(in STW, unless it's somehow already prevented, but in theory could still be 2+ GC racing threads, but that is not a correctness problem, just a wasted time of doing the scan more then once  - you don't really need to elaborate on that)